projects/opensearch.org/package.yml

```diff
@@ -0,0 +1,166 @@
+        else
+          git clone https://github.com/opensearch-project/k-NN .
+        fi
+        git checkout {{version}}.0
+        git reset --hard
+        git submodule foreach --recursive git reset --hard
+        git submodule update --init --recursive
+      working-directory: k-NN
+
+    # workarounds for m1 build. see: https://github.com/opensearch-project/k-NN/blob/main/DEVELOPER_GUIDE.md#extra-setup-for-mac-m1-machines
+    - run: |
+        sed -i -e 's/-march=native/-mcpu=apple-m1/g' nmslib/similarity_search/CMakeLists.txt
+        sed -i -e 's/-mcpu=apple-a14/-mcpu=apple-m1/g' nmslib/python_bindings/setup.py
+        sed -i -e 's/__aarch64__/__undefine_aarch64__/g' faiss/faiss/utils/distances_simd.cpp
+      if: darwin/aarch64
+      working-directory: k-NN/jni/external
+
+    - run: |
+        IFS=:
+        for p in ${LD_LIBRARY_PATH}; do
+          if [ -e "${p}/libomp.dylib" ]; then
+            libomp_path="${p}/.."
+            break
+          fi
+        done
+        unset IFS
+        [ -z "${libomp_path}" ] && echo "libomp.dylib not found" && exit 1
+
+        sed -i -e "s|/usr/local/opt/libomp/|${libomp_path}/|g" jni/CMakeLists.txt
+        sed -i -e 's/pragma message WARN/pragma message /g' jni/external/nmslib/similarity_search/src/distcomp_scalar.cc
+        export CC=clang
+        export CXX=clang++
+      if: darwin
+      working-directory: k-NN
+
+    # this is recommended in https://github.com/opensearch-project/k-NN/blob/45e9e542aef60ef7073ee726e6ac14dec27bfa04/scripts/build.sh#L91-L94
+    - run: sed -i -e 's/-march=native/-march=x86-64/g' CMakeLists.txt
+      working-directory: k-NN/jni/external/nmslib/similarity_search
+      if: x86-64
+    - run: |
+        cmake . --fresh
+        make
+      working-directory: k-NN/jni
+    - run: |
+        for LIB in *.jnilib; do
+          install_name_tool -add_rpath @loader_path $LIB
+        done
+      working-directory: k-NN/jni/release
+      if: darwin
+    - run: |
+        ./gradlew build --refresh-dependencies -x integTest -x test -DskipTests=true -Dopensearch.version={{version}} -Dbuild.snapshot=false -Dbuild.version_qualifier=
+        ./gradlew publishPluginZipPublicationToZipStagingRepository -Dopensearch.version={{version}} -Dbuild.snapshot=false -Dbuild.version_qualifier=
+        ./gradlew publishPluginZipPublicationToMavenLocal -Dbuild.snapshot=false -Dbuild.version_qualifier= -Dopensearch.version={{version}}
+      working-directory: k-NN
+    - run: |
+        mkdir -p ./build/distributions/lib
+        cp -v ./jni/release/libopensearchknn* ./build/distributions/lib
+        cd ./build/distributions
+        zip -r opensearch-knn-{{version}}.0.zip lib/
+        {{prefix}}/bin/opensearch-plugin install --batch file:`pwd`/opensearch-knn-{{version}}.0.zip
+      working-directory: k-NN
+    - run: echo 'export OPENSEARCH_JAVA_OPTS="-Djava.library.path=$OPENSEARCH_HOME/plugins/opensearch-knn/lib $OPENSEARCH_JAVA_OPTS"' >> opensearch-env
```

Comment:
if it's happening when opensearch boots, then it shouldn't be loading the knn plugin yet. it might be related to this. 

i tried to modify the java library path here so that the native plugin dependencies would be loaded properly. maybe it needs some more paths? hmm...

Reply:
does it maybe need a `runtime.env.OPENSEARCH_JAVA_OPTS:` key?